### ISSUE TITLE
Problems authenticating with token from get_azure_cli_credentials

### ISSUE THREAD

**Issue by isobit** [Azure/azure-data-lake-store-python#255]
### Description

Passing a token retrieved from `azure.common.credentials.get_azure_cli_credentials` to `AzureDLFileSystem` results in an exception:

```
...
  File "/home/josh/code/adls-fuse/_venv/lib/python2.7/site-packages/azure_datalake_store-0.0.36-py2.7.egg/azure/datalake/store/core.py", line 70, in __init__
    self.connect()
  File "/home/josh/code/adls-fuse/_venv/lib/python2.7/site-packages/azure_datalake_store-0.0.36-py2.7.egg/azure/datalake/store/core.py", line 88, in connect
    self.azure = DatalakeRESTInterface(token=self.token, **self.kwargs)
  File "/home/josh/code/adls-fuse/_venv/lib/python2.7/site-packages/azure_datalake_store-0.0.36-py2.7.egg/azure/datalake/store/lib.py", line 274, in __init__
    self.head = {'Authorization': token.signed_session(retry_policy=None).headers['Authorization']}
TypeError: signed_session() got an unexpected keyword argument 'retry_policy'
```

---

### Reproduction Steps

First, log in with `az login`.

```
from azure.common.credentials import get_cli_credentials
from azure.datalake.store.core import AzureDLFileSystem

token, subscription_id = get_cli_credentials()
adlfs = AzureDLFileSystem(token=token, store_name='<store name here>')
```

### Environment summary

**SDK Version:** 0.0.36

**Python Version:** 2.7

**OS Version:** CentOS 7

**Shell Type:** Bash

**Comment by akharit** [comment]
Should be fixed in 0.0.37

**Comment by isobit** [comment]
I can confirm the issue is fixed (for me, at least). Thanks!
